ocl/fhir/mixins.py

```diff
@@ -0,0 +1,246 @@
+import requests
+from users.models import UserProfile
+
+from fhir.resources.fhirdate import FHIRDate
+from fhir.resources.identifier import Identifier
+
+from fhir.resources.codesystem import CodeSystem
+from fhir.resources.codesystem import CodeSystemConcept
+
+from fhir.resources.valueset import ValueSet
+from fhir.resources.valueset import ValueSetCompose
+from fhir.resources.valueset import ValueSetComposeInclude
+from fhir.resources.valueset import ValueSetComposeIncludeConcept
+
+from fhir.resources.conceptmap import ConceptMap
+from fhir.resources.conceptmap import ConceptMapGroup
+from fhir.resources.conceptmap import ConceptMapGroupElement
+from fhir.resources.conceptmap import ConceptMapGroupElementTarget
+
+__author__ = 'davetrig'
+
+
+class BaseFhirMixin(object):
+    def get_from_api(self, url):
+
+        #profile = UserProfile.objects.get(mnemonic=username)
+        #api_url = 'http://api.openconceptlab.org:8000'
+        #api_token = profile.user.auth_token.key
+
+        #api_url = 'http://api.openconceptlab.org:8000'
+        api_url = 'http://localhost:8000'
+        api_token = '891b4b17feab99f3ff7e5b5d04ccc5da7aa96da6'
```

Comment:
This should get the API token from the user's request, if present. If there is no API token in the request, then no API token should be used (which means the request may return unauthorized)

Reply:
Please see e.g. https://github.com/OpenConceptLab/oclapi/blob/master/ocl/manage/views.py#L95